INA219.cpp

```diff
@@ -278,8 +292,8 @@ void INA219::write16(t_reg a, uint16_t d) const {
     Wire.send(temp); // write data lobyte;
   #endif
 
-  Wire.endTransmission(); // end transmission
-  delay(1);
+  return Wire.endTransmission(); // end transmission
+//RMW removing delay  delay(1);
```

Comment:
Should keep delay. Store return value of Wire.endTransmission(), delay(1) and return stored value

Reply:
I'm not super familiar with pull request protocol. Are you asking that I make the changes on my side and then submit a separate pull request or is there an easier way?
As for the delay removals - I am using this library from within a FreeRTOS task and have had some troubles with delay vs. vTaskDelay in that RTOS - and other library implementations I've found don't do the delay either - so it seemed a win/win on that front. Do you know that this delay and the other one(s) are required per the spec?

Reply:
Hello,
For the pull request change there is a suggestion here on how to commit changes : https://stackoverflow.com/questions/7947322/preferred-github-workflow-for-updating-a-pull-request-after-code-review

Reply:
EXCELLENT. Thanks so much for the stackoverflow link. I'll make changes and push. 
As for the delay, I really like your idea of using a switch for FreeRTOS. I'll put that in my new version in a bit. Thanks for the guidance and discussion.

Reply:
I have thinked about the solution. The code should not break if FreeRTOS is not installed.
An idea could be to setup a build time define or to uncomment a line
something like 
```
// uncomment following if using FreeRTOS
//#define USE_FREERTOS

#if defined (USE_FREERTOS)
#include <Arduino_FreeRTOS.h>
#define _delay_ms(ms) _vTaskDelay(ms)
#endif
````

and change all delay() in code by _delay_ms()

Reply:
I'm good in concept. A few notes though:
* I commented below about the inconsistency in the ESP32 redefinition of _delay_ms(ms) to delayMicroseconds(ms) -- should that be a different redefinition or should it be delayMicroseconds(ms/1000) since we're talking milliseconds for delay() and vTaskDelay() ?
* I hesitate to suggest bigger changes, but I'm struggling a little with making #define items in the .cpp file when I've almost always done that type of thing in header files. But getting consistent on that would require moving a handful of lines from .cpp to .h ... thoughts?

Reply:
BTW - the reason the .h file matters in this case is that as the consumer of the library, I don't want to actively go in and change your library files since that goofs up re-build consistency and for upgrading the library seamlessly. I'd much rather do a #define USE_FREERTOS just before including your .h file instead and have that cause the redefinition of _delay_ms()
And in similar ways, the debug item is a bit problematic to use. By you defining it as '0', you prevent an outside user from defining it as '1' and require them to modify your header. I'd rather see the use of #ifdef INA219_DEBUG around the debug code so that an outside user can #define INA219_DEBUG and cause your debug stuff to show up. I can propose that change in this upcoming push too if you're ok with it the idea. It would be a smooth way to handle it.

Reply:
Hey there - one more comment - I was just doing testing and I've run into a problem with using vTaskDelay so early in startup/setup. So, I think I'm going to be conservative and back off from the delay change for FreeRTOS. Sorry for the "thrash" here but I'll go do my homework on that topic before I make a separate pull request. Cool?